<launch>
  <arg name="gui" default="true" />

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <!--param name="image_width" value="1280" /-->
    <!--param name="image_height" value="720" /-->
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="frame_rate" type="double" value="30.0" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>

  <node name="geometry" pkg="camera_monitoring" type="nd_geometry.py">
    <param name="threshold" type="int" value="150" />
    <param name="image" type="str" value="/usb_cam/image_raw"/>
  </node>

  <!-- <node name="moments" pkg="camera_monitoring" type="nd_moments.py">
    <param name="threshold" type="int" value="230" />
    <param name="image" type="str" value="/usb_cam/image_raw" />
  </node> -->

  <!-- <node name="plot_nit" pkg="rqt_plot" type="rqt_plot" args="/usb_cam/geometry/minor_axis /usb_cam/moments/minor_axis" if="$(arg gui)" /> -->
  <node name="plot_nit" pkg="rqt_plot" type="rqt_plot" args="/usb_cam/geometry/minor_axis" if="$(arg gui)" />

  <!-- supervise the laser power and speed, tell if the laser is on and if the robot is running -->
  <node name="supervisor" pkg="camera_monitoring" type="nd_laser_supervisor.py" respawn="true" />

</launch>
